dask/blockwise.py

```diff
@@ -268,6 +228,10 @@ def __init__(
             self.concatenate = concatenate
         self.new_axes = new_axes or {}
 
+        # No IO subgraph allowed in `Blockwise`.
+        # Use `BlockwiseIO` to include IO.
+        self.io_name = None
+
```

Comment:
I'm curious why we need this

Reply:
Right - I think this just allows us to avoid duplicating code in `BlockwiseIO` (that is already in `Blockwise`) in places like `__dask_distributed_pack__`.  In some places, we would need to do an instance check to decide if we need to handle the "io-subgraph" if we don't have a consistent attribute like this.

This is probably not a great solution to the code-duplication problem, so feel free to suggest something else :)

Reply:
Okay - The only reason for this is so that we can use the `io_name` attribute in `_optimize_blockwise` without needing to check that we are dealing with a `BlockwiseIO` layer.  I added this information to the comment... Feel free to object to this approach :)